Userland/Libraries/LibJS/Runtime/MathObject.cpp

```diff
@@ -183,27 +183,53 @@ JS_DEFINE_NATIVE_FUNCTION(MathObject::trunc)
 // 21.3.2.30 Math.sin ( x ), https://tc39.es/ecma262/#sec-math.sin
 JS_DEFINE_NATIVE_FUNCTION(MathObject::sin)
 {
+    // 1. Let n be ? ToNumber(x).
     auto number = TRY(vm.argument(0).to_number(global_object));
-    if (number.is_nan())
+    // 2. If n is NaN, n is +0𝔽, or n is -0𝔽, return n.
+    if (number.is_nan() || number.is_positive_zero() || number.is_negative_zero())
+        return number;
+
+    // 3. If n is +∞𝔽 or n is -∞𝔽, return NaN.
+    if (number.is_infinity())
         return js_nan();
+
+    // 4. Return an implementation-approximated Number value representing the result of the sine of ℝ(n).
     return Value(::sin(number.as_double()));
```

Comment:
These fixes are here since I would like to test `Math.sin(-Infinity)` since with my tests it produces a "negative" NaN bit pattern, which is a good test case.

However this does indicate that probably more of the Math functions do not behave as they should with virtualization disabled. However it might be better to solve this on the LibC level so I will not go further than these 3 simple functions.